elyra/pipeline/processor.py

```diff
@@ -605,3 +608,31 @@ def _get_volume_mounts(self, operation: Operation) -> Optional[Dict[str, str]]:
                 formatted_mount_path = f"/{mount_path.strip('/')}"
                 volume_mounts_valid[formatted_mount_path] = pvc_name
         return volume_mounts_valid
+
+    def _get_kubernetes_secrets(self, operation: Operation) -> Optional[List[Tuple]]:
+        """
+        Loops through an Operation's kubernetes secrets to strip whitespace
+        and re-format as a tuple.
+
+        :param operation: the operation to check for secrets
+        :return: tuple of env var name, secret name, and secret key
+        """
+        valid_secrets = []
+        secrets = operation.component_params.get(KUBERNETES_SECRETS)
```

Comment:
Ideally the 'client' (in this case the processor, but the CLI will need this too because this PR introduces a new dependency that `elyra-pipeline describe` needs to expose) shouldn't have to deal with decomposing a KV-pair into its parts. Code maintenance would be a lot easier, if (for example) this call would return `list(dict)` or `list(some-class)`. The dict or class would encapsulate only valid definitions because there is nothing client can do with an invalid definition.